### PR TITLE
added eta_ome filtering

### PR DESCRIPTION
**Pull request by joelvbernier**
Added map filtering to address artifacts I was seeing in recent data (details to follow).

### CODE REVIEW

File: hexrd/findorientations.py [hunk 1]
```diff
@@ -552,6 +591,47 @@ def generate_eta_ome_maps(cfg, hkls=None, save=True):
...
+
+        # filter
+        # !!! False/None: only row median subtraction
+        #     True: use default
+        #     scalar: stdev for filter in pixels
+        # ??? simplify this behavior
+        if filter_stdev:
+            if isinstance(filter_stdev, bool):
+                filter_stdev = filter_stdev_DFLT
+            pf = -gl_filter(pf, filter_stdev)
```
Comment (psavery) [inline hexrd/findorientations.py:628]: It doesn't matter very much, but you might be able to modify `pf` in place via
`pf[:] = -gl_filter(pf, filter_stdev)`

File: hexrd/config/findorientations.py [hunk 2]
```diff
@@ -235,3 +235,10 @@ def file(self):
     @property
     def threshold(self):
         return self._cfg.get('find_orientations:orientation_maps:threshold')
+
+    @property
+    def filter_maps(self):
+        return self._cfg.get('find_orientations:orientation_maps:filter_maps',
+                             default=False)
```
Comment (psavery) [inline hexrd/config/findorientations.py:242]: We should add this as an option in hexdgui I am guessing?

File: hexrd/findorientations.py [hunk 3]
```diff
@@ -57,35 +59,40 @@ def _process_omegas(omegaimageseries_dict):
 
 
 def label_spots(this_map, method, method_kwargs):
-    # !!! need to remove NaNs from map in case of eta gaps?
-    this_map[np.isnan(this_map)] = 0.
+    # !!! need to remove NaNs from map in case of eta gaps
+    # !!! doing offset and truncation by median value now
+    nan_mask = np.isnan(this_map)
+    med_val = np.median(this_map[~nan_mask])
+    this_map[nan_mask] = med_val
```
Comment (psavery) [inline hexrd/findorientations.py:66]: Do you know if we ever confirmed that `this_map` here is a deep copy, so these modifications aren't getting pushed on to the indexing?

File: hexrd/config/findorientations.py (hunk 2 above)
Comment (joelvbernier) [reply hexrd/config/findorientations.py:242]: yeah, we should update the widget.  In the filtering there are two options:

1. row-wise median filtering to get rid of the streaking artifacts; and
2. the above plus filtering with `gauss_laplace` with a user-specified stdev

File: hexrd/findorientations.py (hunk 3 above)
Comment (joelvbernier) [reply hexrd/findorientations.py:66]: I want the changes to persist to the indexing.

File: hexrd/findorientations.py (hunk 1 above)
Comment (joelvbernier) [reply hexrd/findorientations.py:628]: We can probably use the `output` kwarg in `gauss_laplace` 

https://docs.scipy.org/doc/scipy-0.14.0/reference/generated/scipy.ndimage.filters.gaussian_laplace.html#scipy.ndimage.filters.gaussian_laplace

Comment (joelvbernier) [reply hexrd/findorientations.py:628]: have you tried the `[:]` syntax?

Comment (psavery) [reply hexrd/findorientations.py:628]: > have you tried the `[:]` syntax?

Yeah, it works.

**Review by psavery — APPROVED** [review]
LGTM

**Pull request merged**